crowbar_framework/app/models/tempest_service.rb

```diff
@@ -94,6 +94,11 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
 
     role.save
 
+    # Allocate a public IP, tempest needs it
+    tempest_elements, tempest_node, tempest_ha_enabled = role_expand_elements(role, "tempest")
```

Comment:
You can actually directly take `all_nodes`, I guess (since there's no HA here).

But if you keep this code, `tempest_node` is a list of nodes, not a node.